src/main/java/com/openlattice/shuttle/Flight.java

```diff
@@ -50,10 +49,15 @@ public Flight(
                     Map<String, EntityDefinition> entityDefinitions,
             @JsonProperty( Constants.CONDITIONS ) Optional<Conditions> condition,
             @JsonProperty( SerializationConstants.ASSOCIATION_DEFINITIONS_FIELD )
-                    Map<String, AssociationDefinition> associationDefinitions ) {
+                    Optional<Map<String, AssociationDefinition>> associationDefinitions,
+            @JsonProperty( SerializationConstants.NAME) Optional<String> name,
+            @JsonProperty( SerializationConstants.TAGS) Optional<Set<String>> tags
+    ) {
         this.condition = condition;
         this.entityDefinitions = entityDefinitions;
-        this.associationDefinitions = associationDefinitions;
+        this.associationDefinitions = associationDefinitions.orElseGet( HashMap::new );
```

Comment:
Pref `ImmutableMap::of()` since this map is never modified (or even accessed externally) after construction